core/controllers/admin.py

```diff
@@ -875,12 +897,13 @@ def put(self):
 
     @acl_decorators.can_access_admin_page
     def delete(self):
-        username = self.request.get('username')
+        username = self.normalized_request.get('username')
         user_id = user_services.get_user_id_from_username(username)
 
         if user_id is None:
             raise self.InvalidInputException(
                 'User with given username does not exist.')
+        topic_services.deassign_user_from_all_topics(self.user, user_id)
```

Comment:
Why is this here? I don't understand the logic behind it.

Reply:
We need unassign user from all the topics while marking ban so that once we mark them unbanned they don't have topic_manger role.

Reply:
But isn't this the wrong handler? It seems to be the one for _unbanning_ a user.

Reply:
Sorry for the confusion, I've fixed it now!